pkg/render/kubecontrollers/kube-controllers.go

```diff
@@ -242,6 +244,11 @@ func (c *kubeControllersComponent) Objects() ([]client.Object, []client.Object)
 		objectsToDelete = append(objectsToDelete, c.prometheusService())
 	}
 
+	if c.cfg.Terminating {
```

Comment:
Do you need to explicitly delete this if the Installation resource has been terminated? It should be deleted on it's own if it doesn't have a finalizer.

Reply:
> I was surprised by the behavior that these resources did automatically start getting cleaned up because the Installation resource still existed and it just had a DeletionTimestamp.

Maybe it's just perspective, but that didn't / doesn't surprise me, that's what I would have hoped would happen.

I feel like passing back these resources to be cleaned up is a slight change in the point of delete resource logic (at least what I thought the point was). The delete resource logic so far is used to clean up resources no longer needed because of a configuration change, but now it's being used to aid kubernetes cleanup, which we previously left alone. This will at the very least cause a division and possible confusion as to when we pass back these objects to be deleted (Logstorage controller uses finalizers but doesn't clean up the resource like this).

I won't argue to much against this, but in my opinion it would be best to just let kubernetes handle the cleanup naturally.